plugins/lua/buildingplan/planneroverlay.lua

```diff
@@ -368,8 +370,44 @@ function ItemLine:has_filter()
 end
 
 function ItemLine:get_filter_text()
-    -- TODO: make this show the filter's materials instead of "edit filters"
-    return self:has_filter() and '[edit filters]' or '[any material]'
+    local buildingplan = require('plugins.buildingplan')
+    if not buildingplan.hasFilter(
+        uibs.building_type, uibs.building_subtype, uibs.custom_type, self.idx-1)
+    then
+        return '[any material]'
+    end
+    local mats = buildingplan.getMaterialFilter(
+        uibs.building_type, uibs.building_subtype, uibs.custom_type, self.idx-1)
+    local cats = buildingplan.getMaterialMaskFilter(
+        uibs.building_type, uibs.building_subtype, uibs.custom_type, self.idx-1)
+    local enabled_mat_names = {}
+    local enabled_cat_names = {}
+    for name,props in pairs(mats) do
+        local enabled = props.enabled == 'true' and cats[props.category]
+        if enabled then table.insert(enabled_mat_names, name) end
+    end
+    if #enabled_mat_names == 1 then
+        local mat = enabled_mat_names[1]
+        if mat:len() < 17 then
+            return '['..mat..']'
+        else
+            return '['..mat:sub(1,14)..'...]'
+        end
+    elseif #enabled_mat_names > 1 then
+        for cat, _ in pairs(cats) do
+            if cat ~= 'unset' and cats[cat] then
+                table.insert(enabled_cat_names, cat)
+            end
+        end
+        if #enabled_cat_names == 1 then
+            return '[' .. enabled_cat_names[1]:gsub("^%l", string.upper) .. ']'
+        else
+            return '[various materials]'
+        end
+    else
+        -- can result from selecting wood and then toggling "fire safe" etc.
+        return '[impossible filter]'
```

Comment:
you can control the pen to turn this red if you change this function to generate the entire `text` attribute table instead of just the text portion of the `text` attribute